src/RealPOV.Koikatu/RealPOV.cs

```diff
@@ -37,22 +38,18 @@ internal override void EnablePOV()
             }
             else
             {
-                var cameraTarget = GameObject.Find("HScene/CameraBase/Camera/CameraTarget");
-                if(cameraTarget)
+                if(charaQueue == null)
                 {
-                    currentChara = FindObjectsOfType<ChaControl>().OrderBy(x => Vector3.Distance(cameraTarget.transform.position, x.neckLookCtrl.transform.position)).First();
-                    if (currentChara)
-                    {
-                        if(!enteringMessageShown)
-                        {
-                            Logger.LogMessage("Entering POV of character closest to camera center. Move camera to select other characters.");
-                            enteringMessageShown = true;
-                        }
-                    }
-                    else
-                    {
-                        Logger.LogMessage("No characters found. Move camera center close to a character to enter its POV.");
-                    }
+                    charaQueue = new Queue<ChaControl>();
+                    foreach (ChaControl chara in FindObjectsOfType<ChaControl>())
+                        charaQueue.Enqueue(chara);
+                    currentChara = charaQueue.Dequeue();
+                    charaQueue.Enqueue(currentChara);
+                }
+                else
+                {
+                    currentChara = charaQueue.Dequeue();
+                    charaQueue.Enqueue(currentChara);
```

Comment:
Is there any point to this cycling?

Reply:
It makes an infinite loop, seems fine to me. Alternative is a list, index and code to increment it, so this seems simpler.